manim/mobject/geometry.py

```diff
@@ -1554,3 +1556,254 @@ def __init__(self, main_shape, *mobjects, **kwargs):
+
+        angle_1 = angle_of_vector(anchor_angle_1 - inter)
+        angle_2 = angle_of_vector(anchor_angle_2 - inter)
+
+        if other_angle == False:
+            start_angle = angle_1
+            if angle_2 > angle_1:
+                angle_fin = angle_2 - angle_1
+            else:
+                angle_fin = 2 * np.pi - (angle_1 - angle_2)
+        else:
+            start_angle = angle_1
+            if angle_2 < angle_1:
+                angle_fin = -angle_1 + angle_2
+            else:
+                angle_fin = -2 * np.pi + (angle_2 - angle_1)
+
+        Arc.__init__(
+            self,
+            radius=radius,
+            angle=angle_fin,
+            start_angle=start_angle,
+            arc_center=inter,
+            **kwargs
+        )
+        if dot == True:
+            right_dot = Dot(ORIGIN, radius=dot_radius, color=dot_color)
+            dot_anchor = (
+                inter
+                + (self.get_center() - inter)
+                / np.linalg.norm(self.get_center() - inter)
+                * radius
+                / dot_distance
+            )
+            right_dot.move_to(dot_anchor)
+            self.add(right_dot)
+
+
+class RightAngle(VMobject):
+    """Am elbow-type like object representing a right angle of two intersecting lines.
+
+    Parameters
+    ----------
+    line1 : :class:`Line`
+        The first of the intersecting lines.
+    line2 : :class:`Line`
+        The second of the intersecting lines.
+    length : :class:`float`
+        The length of the arms.
+    quadrant : Sequence[:class:`int`]
+        A sequence of two :class:`int` numbers determining which of the 4 quadrants should be used.
+        Possibilities: (1,1), (-1,1), (1,-1), (-1,-1).
+    kwargs
+        Further keyword arguments that are passed to the constructor of :class:`~.VMobject`.
+
+    Examples
+    --------
+
+    .. manim:: RightAngleExample
+
+        class RightAngleExample(Scene):
+            def construct(self):
+                line1 = Line( LEFT, RIGHT )
+                line2 = Line( DOWN, UP )
+                rightangles = [
+                    RightAngle(line1, line2),
+                    RightAngle(line1, line2, length=0.4, quadrant=(1,-1)),
+                    RightAngle(line1, line2, length=0.5, quadrant=(-1,1), stroke_width=8),
+                    RightAngle(line1, line2, length=0.7, quadrant=(-1,-1), color=RED),
+                ]
+                line_list = VGroup( *[VGroup() for k in range(4)] )
+                for k in range(4):
+                    linea = line1.copy()
+                    lineb = line2.copy()
+                    line_list[k].add( linea )
+                    line_list[k].add( lineb )
+                    line_list[k].add( rightangles[k] )
+                line_list.arrange_in_grid(buff=1.5)
+                self.play(
+                    *[ShowCreation(
+                        line_list[k][:2]
+                    ) for k in range(4)]
+                )
```

Comment:
Because this animation doesn't need to be a video, we could change these (and the other examples) to just use `self.add(....)` instead of using animations (i.e. `self.play(...)`, `self.wait(...)`. This will allow the documentation to be generated more quickly.

Reply:
I changed that, thanks!